File: 2/2000s/25s/Maximum_Count_of_Positive_Integer_and_Negative_Integer.cpp
/*
https://leetcode.com/problems/maximum-count-of-positive-integer-and-negative-integer/description/?envType=daily-question&envId=2025-03-12
*/

// TC: O(N), SC: O(1)
class Solution {
public:
    int maximumCount(vector<int>& nums) {
        int sz=nums.size(), countPos=0, countNegs=0, res=0;
        for(int i=0; i<sz; ++i){
            if(nums[i]==0) continue;
            else if(nums[i]>=1) countPos++;
            else countNegs++;
            res=max(countPos, countNegs);
        }
        return res;
    }
};

File: 2/2000s/25s/2529/Maximum_Count_of_Positive_Integer_and_Negative_Integer.py
# https://leetcode.com/problems/maximum-count-of-positive-integer-and-negative-integer/?envType=daily-question&envId=2025-03-12

class Solution:
    def maximumCount(self, nums: List[int]) -> int:
        lengthOfNums=len(nums)
        countPositives=0
        countNegatives=0
        resultantMax=0
        for i in range(0, lengthOfNums, 1):
            if nums[i]==0:
                continue
            elif nums[i]>=1:
                countPositives+=1
            else:
                countNegatives+=1
            resultantMax=max(countPositives, countNegatives)
        return resultantMax
